### PR TITLE
Move cbook._check_in_list() to _api.check_in_list()

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Start addresses the following section of #16181:

> Move all the internal `cbook` functions to (a/multiple) private module(s).
>   ```
>   from matplotlib import _api
> 
>   @_api.deprecated("3.2")
 >  def compare_versions(a, b):
 >  ```
 > reads better than the current
>   ```
>   from matplotlib import cbook
> 
>   @cbook._deprecated("3.2")
>   def compare_versions(a, b):
>   ```
>   and it does not pollute the namespace with the public package name.

This PR does:
- Move cbook._check_in_list() to _api.check_in_list()

### CODE REVIEW

**Review by dopplershift — CHANGES_REQUESTED** [review]
Looks good. Just one minor nit.

**Review by dopplershift — APPROVED** [review]

File: lib/matplotlib/_api.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    values = _values
+    for key, val in kwargs.items():
+        if val not in values:
+            if _print_supported_values:
+                raise ValueError(
+                    f"{val!r} is not a valid value for {key}; "
+                    f"supported values are {', '.join(map(repr, values))}")
+            else:
+                raise ValueError(
+                    f"{val!r} is not a valid value for {key}")
```
Comment (dopplershift) [inline lib/matplotlib/_api.py:34]: Flake8 *really* wants a blank line at the end of the file apparently.

Comment (QuLogic) [reply lib/matplotlib/_api.py:34]: It fits in one line too, I think.

**Comment by QuLogic** [comment]
> Start addresses the following section of #18176:

That seems to be the wrong issue reference?

**Comment by timhoffm** [comment]
Sorry, should have been #16181. Fixed.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**